src/App.tsx

```diff
@@ -574,13 +670,28 @@ function App() {
                   severity={queryErrorMapping[resultStatus].severity as AlertColor}
                 />
               )}
+              {queryFormHasChanged && (
+                <Alert
+                  severity="error"
+                  className="mb-2"
+                  action={
+                    <Button color="inherit" size="small" onClick={() => undoQueryFormChanges()}>
+                      Undo changes
+                    </Button>
+                  }
+                  data-cy="query-form-changed-alert"
+                >
+                  You have edited the query form fields since you&apos;ve submitted your query.
+                  Downloading is disabled until you undo the changes or submit the new query.
+                </Alert>
```

Comment:
I think I see why you can't directly reuse `ErrorAlert` but it would be nice if we could converge on a single component for handling these kinds of errors. Otherwise they will just start stacking up. 

Also:

```suggestion
                <Alert
                  severity="error"
                  className="mb-2"
                  action={
                    <Button color="inherit" size="small" onClick={() => undoQueryFormChanges()}>
                      Undo changes
                    </Button>
                  }
                  data-cy="query-form-changed-alert"
                >
                  You have edited the query form fields since you&apos;ve submitted your query.
                  Downloading is disabled until you undo the changes or submit a new query.
                </Alert>
```

Reply:
> I think I see why you can't directly reuse ErrorAlert but it would be nice if we could converge on a single component for handling these kinds of errors. Otherwise they will just start stacking up.

Agreed. We could refactor `ErrorAlert` to handle more functionality.